File: 조준장/level3/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    station_len = (w * 2) + 1
    start = 1
    
    for station in stations:
        num = station - w - start - 1
        x, y = divmod(num, station_len)
        if x == 0:
            answer += 1
        else:
            answer += x+1
        
        start = station+w+1
    
    if start <= n:
        num = n - start
        x, y = divmod(num, station_len)
        if x == 0:
            answer += 1
        else:
            answer += x+1
        
        
    return answer

File: 이훈일/Programmers/Level3/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int length = 2 * w + 1;
        int count = 0;
        int start = 1;
        for(int i = 0; i < stations.length; i++) {
            int station = stations[i];
            
            if(start < station - w) {
                count += (station - w - start - 1) / length;
                count++;
            }
            
            start = station + w + 1;
        }
        
        if(start <= n) {
            count += (n - start - 1) / length;
            count++;
        }
        
        return count;
    }
}
